frontend/packages/patternfly/src/components/notification-drawer/notification-drawer.tsx

```diff
@@ -0,0 +1,93 @@
+import * as _ from 'lodash';
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+import {
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateSecondaryActions,
+  EmptyStateVariant,
+  Title,
+} from '@patternfly/react-core';
+import {
+  Drawer,
+  DrawerContent,
+  DrawerPanelContent,
+} from '@patternfly/react-core/dist/js/experimental/components/Drawer';
+
+import NotificationDrawerHeading from './notification-drawer-heading';
+import NotificationTypeHeading from './notification-type-heading';
+
+export enum NotificationTypes {
+  info = 'info',
+  warning = 'warning',
+  critical = 'danger',
+  success = 'success',
+  update = 'update',
+}
+
+const emptyState = (toggleExpanded) => (
+  <EmptyState variant={EmptyStateVariant.full} className="co-status-card__alerts-msg">
+    <Title headingLevel="h5" size="lg">
+      No critical alerts
+    </Title>
+    <EmptyStateBody>
+      There are currently no critical alerts firing. There may be firing alerts of other severities
+      or silenced critical alerts however.
+    </EmptyStateBody>
+    <EmptyStateSecondaryActions>
+      <Link to="/monitoring/alerts" onClick={toggleExpanded}>
+        View all alerts
+      </Link>
+    </EmptyStateSecondaryActions>
+  </EmptyState>
+);
+
+const NotificationDrawer: React.FC<NotificationDrawerProps> = ({
+  isExpanded,
+  children,
+  toggleExpanded,
+  alertData = [],
+  updateData = [],
+}) => {
+  const [isAlertExpanded, toggleAlertExpanded] = React.useState<boolean>(true);
+  const [isAvailableUpdateExpanded, toggleAvailableUpdateExpanded] = React.useState<boolean>(false);
+  const criticalAlerts = _.isEmpty(alertData) ? emptyState(toggleExpanded) : alertData;
+  return (
+    <Drawer isExpanded={isExpanded} isInline>
+      <DrawerContent>{children}</DrawerContent>
+      <DrawerPanelContent noPadding>
+        <NotificationDrawerHeading count={alertData.length + updateData.length}>
+          <NotificationTypeHeading
```

Comment:
`Type`, I ponder if `Category` is more clear.  We don't mean anything that already exists, like a `Kind`, or a programmatic `type`.  Not a strong feeling atm, just thinking out loud.